### PR TITLE
[risk=low][RW-7982] Add some UI unit tests for AdminProfile

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/main/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/admin/admin-user-common.tsx [hunk 1]
```diff
@@ -573,7 +573,7 @@ export const InstitutionalRoleOtherTextInput = ({
 }: InstitutionalRoleOtherTextProps) => {
   return affiliation?.institutionalRoleEnum === InstitutionalRole.OTHER ? (
     <TextInputWithLabel
-      dataTestId={'institutionalRoleOtherText'}
```
Comment (jmthibault79) [inline ui/src/app/pages/admin/admin-user-common.tsx:576]: this was the wrong prop

File: ui/src/testing/react-test-helpers.tsx [hunk 2]
```diff
@@ -101,3 +101,11 @@ export const simulateComponentChange = async (
   });
   return await waitOneTickAndUpdate(wrapper);
 };
+
+export const simulateTextInputChange = async (
```
Comment (jmthibault79) [inline ui/src/testing/react-test-helpers.tsx:105]: wee can make many tests more readable with this one ... TODO

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,492 @@
...
+    const wrapper = component();
+    expect(wrapper).toBeTruthy();
+    await waitOneTickAndUpdate(wrapper);
+
+    expect(
+      findDropdown(wrapper, 'initial-credits-dropdown').props.value
+    ).toEqual(TARGET_USER_PROFILE.freeTierDollarQuota);
+
+    const newLimit = 800.0;
+    await simulateComponentChange(
```
Comment (aweng98) [inline ui/src/app/pages/admin/admin-user-profile.spec.tsx:479]: I think checking pre-changed select value would be good to add.

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,492 @@
...
+      BROAD_ADDR_2
+    );
+    expect(findTextInput(wrapper, 'contactEmail').props().value).toEqual(
+      BROAD_ADDR_2
+    );
+    expect(wrapper.find('[data-test-id="email-invalid"]').exists()).toBeFalsy();
+
+    const saveButton = wrapper.find('[data-test-id="update-profile"]');
+    expect(saveButton.exists()).toBeTruthy();
+    expect(saveButton.props().disabled).toBeFalsy();
```
Comment (aweng98) [inline ui/src/app/pages/admin/admin-user-profile.spec.tsx:204]: Is `.exists()` check redundant? If it is missing, doesn't `saveButton.props()` throws exception?

**Review by aweng98 — APPROVED** [review]
LGTM! couple comments.
I don't find unit tests for table toggles here. I guess that's for future work.

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx (hunk 4 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-profile.spec.tsx:204]: if the saveButton is not found, I think this error would be clearer

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx (hunk 3 above)
Comment (jmthibault79) [reply ui/src/app/pages/admin/admin-user-profile.spec.tsx:479]: you mean that I should show this is actually different from the previous?

**Comment by jmthibault79** [comment]
yes the admin table tests will follow

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx (hunk 4 above)
Comment (aweng98) [reply ui/src/app/pages/admin/admin-user-profile.spec.tsx:204]: 👍🏻

File: ui/src/app/pages/admin/admin-user-profile.spec.tsx (hunk 3 above)
Comment (aweng98) [reply ui/src/app/pages/admin/admin-user-profile.spec.tsx]: yes.

**Pull request merged**